### DISCUSSION TITLE
Dual Edge TPU

### DISCUSSION THREAD

**Discussion by MrAnonymous722** [blakeblackshear/frigate#1792] [Ask A Question]
Has anyone had good results using the dual coral device? My machine has an A+E key M.2 slot so its between this or the regular m.2 device. But I'd rather put get the dual one if it works well with Frigate.

edit: Man I sure hope it works, just put in an order for one. I’ll update this with results in case anyone else is curious.
edit 2: Looks like my device's M.2 port is not playing nice with the Coral. I am in the process of returning it and submitted a backorder for a USB device ):

**Comment by ShaBren** [comment]
I use Frigate with a Dual Edge TPU with no issues, but I'm only using a single core because my system doesn't support the PCIe bifurcation required to use both cores (see https://github.com/google-coral/edgetpu/issues/256 for discussion on that issue).

**Reply by MrAnonymous722** [reply]
If it works at all I will be happy. Its been rough trying to find an in stock coral device ):

**Reply by ShaBren** [reply]
If you're not worried about using both cores, then yeah, you should be fine! That's how mine is set up, and I didn't have to do anything special because of it being the dual version.

**Reply by MrAnonymous722** [reply]
For whatever reason my computer doesn't recognize the coral device. I submitted an RMA and put in an order for the USB one instead. There's not much documentation for my computer (Lenovo M910Q) so I don't really know where to go with the M.2 device.

**Reply by bdherouville** [reply]
For the dual TPU you need to install dkms and it will compile modules. 

I plan to buy the dual edge TPU to PCIe adapter.
For the moment only one TPU is recognized by Armbian.

https://github.com/magic-blue-smoke/Dual-Edge-TPU-Adapter

**Comment by doghouselabs** [comment]
you can get the usb accelerator from arrow. they have over 1200 in stock.  $72 with free shipping. i got one & the dual m.2.

https://www.arrow.com/en/products/g950-06809-01/google-corporation

**Reply by MrAnonymous722** [reply]
Have you gotten a tracking number or received it yet? I ordered one 2 days ago but now on their website its estimated to ship in 2023 lol

**Reply by doghouselabs** [reply]
i ordered on the 9th & got it the 15th

**Reply by MrAnonymous722** [reply]
Something about patience is a virtue I guess haha

**Reply by doghouselabs** [reply]
the link i posted says it ships today

**Reply by MrAnonymous722** [reply]
Yeah when I put my order in it was about 9AM on Thursday and it hasn't shipped yet, but it also had a disclaimer at the top of the page about high order volume and delays.

**Reply by doghouselabs** [reply]
hopefully it ships soon for you. mine shipped 2 days after i ordered. the available quantity is only down about 100 since then.

**Reply by MrAnonymous722** [reply]
I'm pretty confident I'll get it some time next week, seeing as others are getting them shipped.
